src/TrueLayer/PaymentsProviders/PaymentsProvidersApi.cs

```diff
@@ -47,5 +48,23 @@ public async Task<ApiResponse<PaymentsProvider>> GetPaymentsProvider(string id)
                 accessToken: authResponse.Data!.AccessToken
             );
         }
+
+        public async Task<ApiResponse<SearchPaymentsProvidersResponse>> SearchPaymentsProviders(SearchPaymentsProvidersRequest searchPaymentsProvidersRequest)
+        {
+            searchPaymentsProvidersRequest.NotNull(nameof(searchPaymentsProvidersRequest));
+
+            ApiResponse<GetAuthTokenResponse> authResponse = await _auth.GetAuthToken(new GetAuthTokenRequest("payments"));
+
+            if (!authResponse.IsSuccessful)
+            {
+                return new(authResponse.StatusCode, authResponse.TraceId);
+            }
```

Comment:
For the future, shall we use a delegating handler, with caching capabilities for client credentials token?

Reply:
Is this something the java lib does? What do you cache exactly? And you do it just to avoid the call?
How long do you cache it for?
Generally speaking, yes we can always improve things using caching mechanisms where possible

Reply:
Yup we have something similar there, in Retrofit/OkHttp language they call it interceptor but conceptually it behaves similarly to the handler. If you want to check, [this](https://github.com/TrueLayer/truelayer-java/blob/main/src/main/java/com/truelayer/java/http/interceptors/AuthenticationInterceptor.java) is how we do it